### ISSUE TITLE
Check side effects of ChimeraTK/DeviceAccess#116 and ChimeraTK/DeviceAccess#117

### ISSUE THREAD

**Issue by killenb** [ChimeraTK/ApplicationCore#153]
Child of #123
Depends on #151

Until now readLatest() and readNonBlocking() the postRead() function has not been called if there was no new data. This is wrong, there might be stuff to do like exception handling.
* In ChimeraTK/DeviceAccess#116 postRead() got a new argument `hasNewData`. This has to be used in each implementation to device if action is required if there is new data.
* Since ChimeraTK/DeviceAccess#117 postRead() and postWrite() can throw. If a doPostXxx overwrite in a decorator delegated the call to _impl->postXxx and does something after that call, it might need a try-catch block, do the action after the postXxx and re-throw if there was an exception. "Might need it" because not always the action is required. If the action should only take place for a successful transfer, the try/catch/re-throw is not needed.

* [ ] Check each doPostRead() and doPostWrite() implementation if the action should always take place in case of no new data and/or in case of an exception.
* [ ] Note in this ticket which places have been changed.

Examples
------------
* If preRead() is empty, and post read swaps the new data into the user buffer, this must not happen because the old user buffer must stay intact => no action if no new data
* If preRead() swaps out the user buffer such that the transfer can modify it, which it does not do because there was no new data, it has to be swapped back to get the user buffer back into place => action always required, even if no new data

**Comment by phako** [comment]
So. we have

- ConsumingFanOut::doPostRead()
- FeedingFanOut::doPostRead()
- FeedingFanOut::doPostWrite()
- FeedingFanOut::doReadTransferAsync()

I think that's it. It needs a couple of tests which I will write a follow-up ticket foir
